### PR TITLE
Refactor GetResourceRequest and GetResourceLimit

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/assign @bsalamat 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/schedulercache/node_info_test.go [hunk 1]
```diff
@@ -185,6 +185,56 @@ func TestResourceAddScalar(t *testing.T) {
...
+				MilliCPU:         4,
+				Memory:           4000,
+				EphemeralStorage: 5000,
+				ScalarResources:  map[v1.ResourceName]int64{"scalar.test/scalar1": 1, "hugepages-test": 2},
+			},
+			resourceList: map[v1.ResourceName]resource.Quantity{
+				v1.ResourceCPU:                      *resource.NewScaledQuantity(4, -3),
+				v1.ResourceMemory:                   *resource.NewQuantity(2000, resource.BinarySI),
+				v1.ResourceEphemeralStorage:         *resource.NewQuantity(7000, resource.BinarySI),
+				"scalar.test/" + "scalar1":          *resource.NewQuantity(4, resource.DecimalSI),
```
Comment (k82cn) [inline pkg/scheduler/schedulercache/node_info_test.go]: just single string

File: pkg/scheduler/schedulercache/node_info.go [hunk 2]
```diff
@@ -202,6 +202,39 @@ func (r *Resource) SetScalar(name v1.ResourceName, quantity int64) {
 	r.ScalarResources[name] = quantity
 }
 
+// SetMaxResource compare ResourceList and take max value for Resource.
```
Comment (k82cn) [inline pkg/scheduler/schedulercache/node_info.go]: 1. s/compare/compares with/
1. s/take/takes/
1. s/Resource/each Resource/

**Comment by wackxu** [comment]
@k82cn Done, PTAL

File: pkg/scheduler/schedulercache/node_info.go [hunk 3]
```diff
@@ -202,6 +202,37 @@ func (r *Resource) SetScalar(name v1.ResourceName, quantity int64) {
 	r.ScalarResources[name] = quantity
 }
 
+// SetMaxResource compares with ResourceList and takes max value for each Resource.
+func (r *Resource) SetMaxResource(rl v1.ResourceList) {
+	if r == nil {
```
Comment (resouer) [inline pkg/scheduler/schedulercache/node_info.go]: Isn't it a nil pointer error?

Comment (wackxu) [reply pkg/scheduler/schedulercache/node_info.go]: Yes

**Review by ravisantoshgudimetla — COMMENTED** [review]
@wackxu Can you squash the commits. This looks good to me unless @resouer has any other concern

**Comment by wackxu** [comment]
@ravisantoshgudimetla Done, PTAL

**Review by ravisantoshgudimetla — APPROVED** [review]
Thanks @wackxu

**Pull request merged**